### ISSUE TITLE
Add routes for export URLs

### ISSUE THREAD

**Issue by adamfranco** [middlebury/coursecatalog#7]
This is related to #6 , but the URL should use the numeric ID for the config so that the URL won't break if the title contains slashes or other special characters.

Also, a route should be added to [`application/library/routes.php` ](https://github.com/middlebury/coursecatalog/blob/master/application/library/routes.php) so that the URL is more clean:

Example:

-  `/admin/export` would be the main screen with a selector for configs 
-  `/admin/export/1` is the detail screen. 
-  `/admin/export/1/history` can be the history screen
-  `/admin/export/1/json/71` would be the current `viewJson` action for a revision 71.
- etc

You can still use `$request->getParam('configId')` in the controller, it will just be mapped from the positional location to a named parameter by the router.

**Comment by cg923** [comment]
Hmm.  So I definitely see the sense in having easier to read routes, and replacing parameters like "Fall+Spring+2018" with the configId.

However, because of the way routes are already set up, changing /export?config=1 to /export/1 means renaming and reworking almost every other route related to the project.  For example, /export/latestrevision, based on the new naming scheme, is triggering /export/config but with a configId of "latestrevision"...

In other words, I am afraid this is going to be quite a complicated fix.  Which is okay and I'm willing to do it, but it might take more time than it's worth.  

Certainly I can at least make the URLs more readable by changing the way parameters are thrown around.  Any thoughts @adamfranco ?

**Comment by adamfranco** [comment]
You should be able to leave the controller totally as-is (no renaming) and just add entries to: [application/library/routes.php](https://github.com/middlebury/coursecatalog/blob/master/application/library/routes.php#L78-L125). The Kurogo URIs at the bottom provide an example of how to mix parameters and tokens.

Documentation for adding routes that match particular syntax: https://framework.zend.com/manual/1.12/en/zend.controller.router.html#zend.controller.router.routes.regex

**Comment by cg923** [comment]
I did look at those docs and was able to make certain routes work, but others were conflicting.  I'll take a closer look, thanks Adam.

**Comment by adamfranco** [comment]
Also note that the order of the routes matters. From the documentation:

> Note: Reverse Matching
> Routes are matched in reverse order so make sure your most generic routes are defined first.

**Comment by cg923** [comment]
Ah.  That was definitely my problem!
